### ISSUE TITLE
New variable resolver documentation is vague

### ISSUE THREAD

**Issue by OndeVai** [serverless/serverless#9163]
The documentation for the new variables resolver (https://www.serverless.com/framework/docs/providers/aws/guide/variables#exporting-a-function) is lacking clarity.

1) The code sample does not even show a return value. Plus why does the function need to make async calls to the config file?
2) The description below left me super confused:
```
In the old legacy resolver (which is deprecated, but stays as default in v2) function receives a reference to the Serverless object containing your configuration.

Notice: Configuration is yet in unresolved state, so any properties configured with variables may still be presented with variables in it
```
3) The other page that references the associated warning from the CLI (https://www.serverless.com/framework/docs/deprecations#new-variables-resolver) (which is also verbose and confusing) does not link to anywhere helpful to learn more. Also, that documentation states 
`It supports very same variable syntax, and is being updated with support for same resolution sources. Still as it has improved internal resolution rules (which leave no room for ambiguity) in some edge cases it may report errors on which old parser passed by.`

However, the syntax is very different. Also, there's some grammar issues :)

**Comment by medikoo** [comment]
@OndeVai thanks for opening that.

> The code sample does not even show a return value.

Example focuses on showing things that are not obvious. I imagine it's obvious that function is expected to return the resolved value.

Anyway I don't say that example cannot be improved. We'll definitely open for PR's that improves it.

 > why does the function need to make async calls to the config file?

Config properties may be behind variables of which resolution can be async. This function guarantess you'll get the final resolved value for config property.

> The description below left me super confused:

Can you describe what's confusing (?)

> However, the syntax is very different. 

Syntax stayed same, why do you assume is different?

> Also, there's some grammar issues :)

We're open for PR that improves it, can you open one?

**Comment by OndeVai** [comment]
Hi @medikoo,
Thanks for following up.

> Can you describe what's confusing (?)

I guess the wording of "It supports very same variable syntax" is vague to me. After implementing the new resolver I think that sentence is referring to the syntax when referencing the js file in serverless.yml.

Syntax stayed same, why do you assume is different?

The syntax of the function itself is very different --hence my confusion.

> Also, there's some grammar issues :)

There's some missing "the's" some other articles. --No biggie, but I thought I'd mention it.

> We're open for PR that improves it, can you open one?

I will try to carve some time this weekend to submit a PR.

Thanks for the great work --you guys are awesome BTW!

**Comment by medikoo** [comment]
> I think that sentence is referring to the syntax when referencing the js file in serverless.ym

Yes, it refers to that, and that syntax stayed same hence _"It supports very same variable syntax"_

> The syntax of the function itself is very different --hence my confusion.

What's different is function _signature_ (arguments it takes). Technically in both case we speak about JS functions, and in that context syntax stayed same.

Anyway we know we have missings in documentation, and that it requires improvements. Currently we're figuring out resources to address that
